### ISSUE TITLE
数据集问题请教，请教对比算法DGrec的数据处理

### ISSUE THREAD

**Issue by ahukmr** [Xiaolong-Liu-bdsc/KG-diverse#1]
作者您好，我想请教下，在您的论文中比较了对比算法DGCN和DGrec，这两个模型都用到了item的类别属性信息，想请问您的数据集中item的类别属性信息是怎么处理的，我好像在数据集中没有看到item的类别属性信息文件，想请教您这两个算法数据集是怎么处理的，谢谢！

**Comment by Xiaolong-Liu-bdsc** [comment]
KG里没有item的类别属性信息, 我们采用其他特征来取代的。
例如DGRec的reweight W_{C_{(i)}} in Eq 12, 我们用item的frequency来替代的

**Comment by Xiaolong-Liu-bdsc** [comment]
> 作者您好，我想请教下，在您的论文中比较了对比算法DGCN和DGrec，这两个模型都用到了item的类别属性信息，想请问您的数据集中item的类别属性信息是怎么处理的，我好像在数据集中没有看到item的类别属性信息文件，想请教您这两个算法数据集是怎么处理的，谢谢！

KG里没有item的类别属性信息, 我们采用其他特征来取代的。
例如DGRec的reweight W_{C_{(i)}} in Eq 12, 我们用item的frequency来替代的.本来是按照category的frequency来降低training的weight, 没有category的时候我们就按照item frequency来降低这个weight。选neighbor的时候，submodular function用的feature similarity。

**Comment by ahukmr** [comment]
非常感谢您的回答，请问submodular function用的feature similarity,这里具体是对所有物品之间做相似度计算吗，不知道能否提供下关于DGrec数据处理和子模选择部分的代码呀？因为我最近也想使用这个模型，但是dataloader和子模选择部分的代码处理起来有点问题，如果可以希望能学习下您的做法，非常感谢！

**Comment by Xiaolong-Liu-bdsc** [comment]
> 非常感谢您的回答，请问submodular function用的feature similarity,这里具体是对所有物品之间做相似度计算吗，不知道能否提供下关于DGrec数据处理和子模选择部分的代码呀？因为我最近也想使用这个模型，但是dataloader和子模选择部分的代码处理起来有点问题，如果可以希望能学习下您的做法，非常感谢！

是的。 https://github.com/YangLiangwei/DGRec/blob/main/models/layers.py 47-52行 如果没有category的话 代码会call similarity_matrix来计算相似度
您可以发邮件给我 我给你发送数据处理的代码~

**Comment by ahukmr** [comment]
> > 非常感谢您的回答，请问submodular function用的feature similarity,这里具体是对所有物品之间做相似度计算吗，不知道能否提供下关于DGrec数据处理和子模选择部分的代码呀？因为我最近也想使用这个模型，但是dataloader和子模选择部分的代码处理起来有点问题，如果可以希望能学习下您的做法，非常感谢！
> 
> 是的。 https://github.com/YangLiangwei/DGRec/blob/main/models/layers.py 47-52行 如果没有category的话 代码会call similarity_matrix来计算相似度 您可以发邮件给我 我给你发送数据处理的代码~

非常感谢，我在您的主页没看到您的邮箱，或者我给您我的邮箱，1310969321@qq.com,还麻烦您分享下您数据处理的代码，特别是对比算法DGrec在没有类别信息的数据集下运行的数据处理和模型处理的代码，再次谢谢您！

**Comment by Ty5q** [comment]
作者您好，请问可以给我也发一份数据处理的代码吗？十分感谢！
邮箱：1976573947@qq.com

**Comment by Krsy22** [comment]
作者您好，README中只放了对于music_small的参数，可以发一份其余数据集的最优参数嘛
